packages/storage-plugin/tests/storage-for-feature.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { BrowserModule } from '@angular/platform-browser';
+import { RouterModule, Router } from '@angular/router';
+import { State, NgxsModule, Store, provideStates } from '@ngxs/store';
+import { Component, Injectable, NgModule, NgZone } from '@angular/core';
+import { freshPlatform, skipConsoleLogging } from '@ngxs/store/internals/testing';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { APP_BASE_HREF } from '@angular/common';
+
+import { NgxsStoragePluginModule, withStorageFeature } from '..';
+
+describe('forFeature', () => {
+  interface CounterStateModel {
+    count: number;
+  }
+
+  @State<CounterStateModel>({
+    name: 'counter',
+    defaults: { count: 0 }
+  })
+  @Injectable()
+  class CounterState {}
+
+  interface BlogStateModel {
+    pages: number[];
+  }
+
+  @State<BlogStateModel>({
+    name: 'blog',
+    defaults: { pages: [] }
+  })
+  @Injectable()
+  class BlogState {}
+
+  @Component({ template: 'This is blog' })
+  class BlogComponent {}
+
+  @NgModule({
+    imports: [RouterModule.forChild([{ path: '', component: BlogComponent }])],
+    declarations: [BlogComponent],
+    providers: [provideStates([BlogState], withStorageFeature([BlogState]))]
+  })
+  class BlogModule {}
+
+  @Component({
+    selector: 'app-root',
+    template: '<router-outlet></router-outlet>'
+  })
+  class TestComponent {}
+
+  @NgModule({
+    imports: [
+      BrowserModule,
+      RouterModule.forRoot(
+        [
+          {
+            path: 'blog',
+            loadChildren: () => BlogModule
+          }
+        ],
+        {
+          initialNavigation: 'disabled'
+        }
+      ),
+      NgxsModule.forRoot([CounterState]),
+      NgxsStoragePluginModule.forRoot({
+        key: [CounterState]
+      })
```

Comment:
We also need to test the case where the root plugin has been initialised with no keys present (therefore every state is stored), and then the `blog` route is loaded, which will set a key and engine. Will the existing states be stored? I'm not sure if the current logic handles this scenario. 

I think the expectation would be that all states would be stored in the default key, and that explicitly specified feature states would be stored separately. 
Maybe we need to discuss what would happen for any other feature states... What is the current behaviour for lazy feature states?